### PR TITLE
[Python] Added Allowed Collision Matrix to planning scene and optional planning scene during planning

### PR DESCRIPTION
**Pull request by JensVanhooydonck**
### Description

Added the ACM to the planning scene object. This allows setting the Allowed Collision Object Matrix.
Also added an optional planning_scene in the Plan function so another planning can be used then the current planning scene. (Can be used as a base for a basic python Task manager)

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit2/blob/main/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://moveit.picknik.ai/humble/doc/examples/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit2/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_py/src/moveit/moveit_ros/moveit_cpp/planning_component.cpp [hunk 1]
```diff
@@ -73,7 +74,7 @@ plan(std::shared_ptr<moveit_cpp::PlanningComponent>& planning_component,
     if (solution_selection_function && stopping_criterion_callback)
     {
       return planning_component->plan(*const_multi_plan_parameters, std::ref(*solution_selection_function),
-                                      *stopping_criterion_callback);
+                                      *stopping_criterion_callback, planning_scene);
     }
     else if (solution_selection_function)
     {
```
Comment (peterdavidfagan) [inline moveit_py/src/moveit/moveit_ros/moveit_cpp/planning_component.cpp:80]: Line 81 is missing the planning_scene argument.

**Review by peterdavidfagan — COMMENTED** [review]
These changes need to be tested but on a high-level the changes look good to me. Once I have patched the issue with parameters in the tutorials I'll look to test and approve changes.

File: moveit_py/src/moveit/moveit_ros/moveit_cpp/planning_component.cpp (hunk 1 above)
Comment (JensVanhooydonck) [reply moveit_py/src/moveit/moveit_ros/moveit_cpp/planning_component.cpp:80]: https://github.com/ros-planning/moveit2/blob/cc7a0d422b294ffc7c01a52df074140bb3c5aa15/moveit_ros/planning/moveit_cpp/src/planning_component.cpp#L211-L212 Does not allow for a planning scene argument.
I can see if i can add that to the function. Or i can throw an exception if there is this combinaiton was added?

**Comment by JensVanhooydonck** [comment]
@peterdavidfagan I've added the return policy and the planning scene argument to line 81.

**Comment by peterdavidfagan** [comment]
Thanks @JensVanhooydonck, looks good to me. Will test on a robot later today.

**Review by peterdavidfagan — APPROVED** [review]

**Review by henningkayser — APPROVED** [review]

**Pull request merged**